src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/services/CaptchaService.java

```diff
@@ -0,0 +1,70 @@
+package com.sublinks.sublinksapi.api.lemmy.v3.user.services;
+
+import com.sublinks.sublinksapi.api.lemmy.v3.user.dto.Captcha;
+import com.sublinks.sublinksapi.api.lemmy.v3.user.repositories.CaptchaRepository;
+import jakarta.transaction.Transactional;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.UUID;
+import javax.imageio.ImageIO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ResponseStatusException;
+
+@Component
+@RequiredArgsConstructor
+public class CaptchaService {
+
+  private final CaptchaRepository captchaRepository;
+
+  @Transactional
+  public Captcha getCaptcha() {
+    Captcha captcha = captchaRepository.findFirst();
+
+    if (captcha == null) {
+      captcha = createCaptcha();
+    }
+
+    captcha.setLocked(true);
```

Comment:
We need a timeout for this otherwise there can be a overflow attack ( in "just" requestion captchas => locking literally everything )